Backend/lib/sessionizer-metrics/src/Lib/EventStream.hs

```diff
@@ -0,0 +1,111 @@
+
+ the GNU Affero General Public License along with this program. If not, see <https://www.gnu.org/licenses/>.
+-}
+
+module Lib.EventStream where
+
+import Kernel.Prelude hiding (at, traceId)
+import Kernel.Streaming.Kafka.Producer.Types (KafkaProducerTools)
+import Kernel.Utils.Common
+import Lib.Kafka.Internal
+import Lib.Types.Event
+import Lib.Types.Payload
+
+triggerEvent ::
+  ( Monad m,
+    Log m,
+    MonadFlow m,
+    MonadReader r0 m,
+    HasFlowEnv m r '["kafkaProducerTools" ::: KafkaProducerTools],
+    HasField "eventStreamMap" r [EventStreamMap]
+  ) =>
+  Event ->
+  m ()
+triggerEvent event = do
+  allEventStream <- asks (.eventStreamMap)
+  let streamNames = filter (elem (eventType event) . eventTypes) allEventStream
+  logDebug $ "my filtered stream map" <> show streamNames
+  forM_ streamNames $ \stream -> do
+    case streamName stream of
+      KAFKA_STREAM -> do
+        let KafkaStream matchedConfig = stream.streamConfig
+        fork "updating in kafka" $ streamUpdates event matchedConfig
+      _ -> logDebug "Default stream"
+
+createEvent :: (MonadGuid m, MonadTime m) => Text -> Text -> Text -> Text -> Text -> Maybe Payload -> m Event
+createEvent personId merchantId eventType deployVersion service payload = do
+  uid <- generateGUID
+  now <- getCurrentTime
+  let ev =
+        Event
+          { id = uid,
+            traceId = Nothing,
+            sessionId = Nothing,
+            personId = personId,
+            merchantId = merchantId,
+            deploymentVersion = deployVersion,
+            at = now,
+            eventType = eventType,
+            subType = Nothing,
+            primaryId = Nothing,
+            service = service,
+            triggeredBy = System,
+            payload = payload
+          }
+  return ev
+
+createRidePayload :: Text -> Text -> UTCTime -> UTCTime -> RidePayload
+createRidePayload rideId rideStatus createdAt updatedAt =
+  RidePayload
+    { rId = rideId,
+      st = rideStatus,
+      cAt = createdAt,
+      uAt = updatedAt
+    }
+
+createBookingPayload :: Text -> Text -> UTCTime -> UTCTime -> BookingPayload
+createBookingPayload bId bSt createdAt updatedAt =
+  BookingPayload
+    { bId = bId,
+      st = bSt,
+      cAt = createdAt,
+      uAt = updatedAt
+    }
+
+createSearchPayload :: Text -> Maybe Bool -> UTCTime -> SearchPayload
+createSearchPayload searchId autoAssigned createdAt =
+  SearchPayload
+    { sId = searchId,
+      auAssign = autoAssigned,
+      cAt = createdAt
+    }
+
+createQuotePayload :: Text -> Maybe Text -> Text -> Text -> UTCTime -> Maybe UTCTime -> QuotePayload
+createQuotePayload quoteId status seachReqId searchTryId createdAt updatedAt =
+  QuotePayload
+    { qId = quoteId,
+      st = status,
+      searchReqId = seachReqId,
+      searchTryId = searchTryId,
+      cAt = createdAt,
+      uAt = updatedAt
+    }
+
+createEstimatePayload :: Text -> Text -> Text -> UTCTime -> EstimatePayload
```

Comment:
all these functions can have return type as Payload instead of specific payload ?

Reply:
Done